.github/actions/upgrade-indexer/common.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+FILES_OLD="/usr/share/wazuh-indexer/plugins/opensearch-security/securityconfig"
+FILES_NEW="/etc/wazuh-indexer/opensearch-security"
+declare -A files_old
+declare -A files_new
+PACKAGE_NAME="${1}"
+MAJOR_MINOR_RELEASE=$((${2}))
+REFERENCE_VERSION=43
+
+# Check the system to differ between DEB and RPM
+function check_system() {
+
+    if [ -n "$(command -v yum)" ]; then
+        sys_type="rpm"
+    elif [ -n "$(command -v apt-get)" ]; then
+        sys_type="deb"
+    else
+        echo "Error: could not detect the system."
+        exit 1
+    fi
+
+}
+
+# Checks the version of Wazuh with 4.3 version, where path is different.
+function check_version() {
+    if [ -z "${MAJOR_MINOR_RELEASE}" ]; then
+        echo "Error: second argument expected"
+        exit 1
+    fi
+
+    if [ "${MAJOR_MINOR_RELEASE}" -gt "${REFERENCE_VERSION}" ]; then
+        FILES_OLD="${FILES_NEW}"
+        echo "New path detected (/etc)."
+    else
+        echo "Old path detected (/usr/share)."
+    fi
+}
+
+# Compare the arrays, the loop ends if a different checksum is detected
+function compare_arrays() {
+
+    for i in "${!files_old[@]}"; do
+        echo "Comparing $i file checksum..."
+        echo "Old: ${files_old[$i]}"
+        echo "New: ${files_new[$i]}"
+        if [[ "${files_old[$i]}" == "${files_new[$i]}" ]]; then
+            echo "${i} - Same checksum."
+        else
+            echo "${i} - Different checksum."
+            return 1
+        fi
+    done
+    return 0
+}
+
+# Steps before installing the RPM release package.
+function preinstall_indexer_release() {
```

Comment:
This function adds a production repository. It doesn't preinstall anything.

Reply:
Done. Changed by `add_production_repository`